content/recipes/authoring/kata-snippets/submission-tests.md

```diff
@@ -0,0 +1,88 @@
+  - Input mutation by the user solution which can affect the input passed to the reference solution, or make assertion messages confusing
+  - Incorrect implementation of the reference solution leading to rejection of valid user solutions
+- **The reference solution should not be revealed to the user.** When an assertion fails or the test suite crashes, some testing frameworks print fragments of source code which caused the failure to the console. It may happen that such printed failure messages or stack traces expose information about the solution which should not be revealed.
+- **The reference solution shouldn't be accessible to the user solution.** It should not be possible to call the reference solution directly, or implement the user solution as an alias or wrapper around the reference solution. The reference solution should be completely inaccessible outside the submit tests. In particular, it should not be a global and/or public function. Check the [reference page for your language](/languages/) to see how to prevent this problem in your tests.
+
+
+## Fixed Tests
+
+Fixed tests are tests with predetermined inputs and outputs, and do not change between test runs.
+
+- **Each requirement should have a corresponding fixed test.** Every aspect of the specification in the kata description should be explicitly tested with at least one, possibly more, dedicated and properly labeled assertion(s). Fixed tests are usually easier to debug, so ideally a fixed test should fail before the corresponding scenario is tested with random tests.
+- **Tests should check the solution with edge cases and cases that require special handling.** For example, empty arrays and empty strings should be covered by fixed tests unless the kata description _explicitly_ states that such inputs need not be considered.
+
+
+## Random Tests
+
+Random tests are uncommon in "real life" coding and are somewhat specific to Codewars. They are required by many kata to reject solutions based on input probing, hard-coding, and other workarounds. The goal of random tests is to make the inputs unpredictable so that only proper solutions may pass.
+
+- **Keep the amount of random tests reasonable.** If random tests are not used to verify performance, keep their amount small enough to guarantee sufficient coverage. In some cases it's not necessary to have dozens, or even hundreds of random tests, and several of them would be enough. Do not create random tests just for the sake of their amount. However, if you believe that a large amount of random tests helps you to detect invalid solutions which are not caught by fixed tests, you should try and find a pattern in what types of cases fail particularly often and extract them to the corresponding fixed tests.
+- Random tests can be limited to certain types of inputs not covered with fixed tests in the whole range. They should generate test cases for all scenarios which cannot be completely tested with fixed tests.
```

Comment:
this § has to be rewritten.
* I don't understand the sentences construction
* I don't get the underlying intent
* some parts even seem in contradiction with some others, to me...

I cannot suggest anything since I'm not even sure what this means... :o (I just hope I'm not the one who wrote that x) )